bundles/org.eclipse.jface/src/org/eclipse/jface/viewers/StructuredViewer.java

```diff
@@ -814,6 +815,11 @@ public void run() {
 	 * @see #addPostSelectionChangedListener(ISelectionChangedListener)
 	 */
 	protected void firePostSelectionChanged(final SelectionChangedEvent event) {
+		// do not inform client listeners on ExpandableNode selection
+		if (event.getSelection() instanceof StructuredSelection sel
+				&& sel.getFirstElement() instanceof ExpandableNode) {
```

Comment:
I have missed this PR but wanted to note that selections can have multiple elements selected and that case is not handled.

Also I think the node must be filtered at the creation of the selection event instead of this protected method.

FYI @mickaelistria